### PR TITLE
Update the FindMKL module to squash warnings

### PR DESCRIPTION
**Pull request by teonnik**
The module is based on https://github.com/eth-cscs/cmake-recipes/blob/master/FindMKL.cmake .

### CODE REVIEW

File: cmake/FindMKL.cmake [hunk 1]
```diff
@@ -175,135 +127,183 @@ endif()
...
+#
 __mkl_find_library(MKL_SEQ_LIB mkl_sequential)
-if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND NOT APPLE)
+if(NOT APPLE AND (CMAKE_C_COMPILER_ID STREQUAL "GNU" OR
+                  CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR
+                  CMAKE_Fortran_COMPILER_ID STREQUAL "GNU"))
     __mkl_find_library(MKL_OMP_LIB mkl_gnu_thread)
 else()
     __mkl_find_library(MKL_OMP_LIB mkl_intel_thread)
 endif()
```
Comment (rasolca) [inline cmake/FindMKL.cmake:200]: I'm wondering about Clang openMP... Does anyone know if it is compatible with any of the MKL library?

**Comment by teonnik** [comment]
bors try

File: cmake/FindMKL.cmake (hunk 1 above)
Comment (teonnik) [reply cmake/FindMKL.cmake:200]: I am not sure... unless someone chimes in, I suggest we address it when/if we encounter issues.

**Review by rasolca — APPROVED** [review]
LGTM

**Pull request merged**